### ISSUE TITLE
Change Command: next_boss

### ISSUE THREAD

**Issue by harbingerofme** [harbingerofme/DebugToolkit#88]
**Which Command**
`next_boss`

**What should it do?**
Actually work correctly for any master. rewrite the entire stuff with some more aggressive hooks.

**Comment by SChinchi** [comment]
There are still some issues with this command:

1. If the elite is not specified, `CurrentRun.nextElite.eliteIndex` will throw null. This forces the command to require an elite.
2. Passing -1/None for the elite will technically return null from `EliteCatalog.GetEliteDef()`. Ideally we want to differentiate between none and an invalid tier so we can log an error for the latter only.
3. The director card cost is modified, effectively corrupting that forever. It's not a big problem as the director cards aren't used for anything else yet, but it's probably better if  we override the director's credits instead.
4. `CombatDirector.SetNextSpawnAsBoss` is called in 3 occasions:
    * teleporter boss
    * simulacrum boss wave: we want the max enemy count to be 10 here instead of 6
    * void locus signals: skip the hook since an unsuspecting user probably does not intend to override this spawn

I can make a PR addressing all of the above unless you have some other suggestions.

**Comment by SChinchi** [comment]
Yes, these are the two commands that deal with elite inputs. What I'm thinking is creating a `GetEliteFromPartial` which will return -2 for invalid, -1 for none, and the index otherwise. Then we can report error/do nothing/do elite stuff accordingly.

**Comment by xiaoxiao921** [comment]
Sounds good

**Comment by SChinchi** [comment]
I'll make a PR which addresses issues 1 and 2 from my previous post. Let me know if you have any thoughts on 3 or 4. They don't seem urgent anyway.

**Comment by xiaoxiao921** [comment]
Agree for 3 / 4
